plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSTopLevelElementValidator.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.validation.validators;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.n4js.n4JS.FunctionDeclaration;
+import org.eclipse.n4js.n4JS.Script;
+import org.eclipse.n4js.n4JS.ScriptElement;
+import org.eclipse.n4js.n4JS.Statement;
+import org.eclipse.n4js.validation.AbstractN4JSDeclarativeValidator;
+import org.eclipse.n4js.validation.IssueCodes;
+import org.eclipse.n4js.validation.JavaScriptVariant;
+import org.eclipse.n4js.validation.JavaScriptVariantHelper;
+import org.eclipse.xtext.validation.Check;
+import org.eclipse.xtext.validation.EValidatorRegistrar;
+
+import com.google.inject.Inject;
+
+/**
+ * A validator to issue error messages on top-level elements based on the {@link JavaScriptVariant} in use.
+ */
+public class N4JSTopLevelElementValidator extends AbstractN4JSDeclarativeValidator {
```

Comment:
move the one check method inside this class to `N4IDLVersionValidator`, delete this class, and then rename `N4IDLVersionValidator` to `N4IDLValidator`.